django/views/generic/dates.py

```diff
@@ -687,7 +698,8 @@ def _get_next_prev(generic_view, date, is_previous, period):
         # Snag the first object from the queryset; if it doesn't exist that
         # means there's no next/previous link available.
         try:
-            result = getattr(qs[0], date_field)
+            result = functools.reduce(
```

Comment:
These lines aren't straightforward. Maybe there's an alternative without functools.reduce that would improve readability?  Even so, this could be a single line... we allow up to 119 characters when it improves readability.